### PR TITLE
Prevent Buffer Overflow

### PR DESCRIPTION
**Pull request by hallard**
Not sure buffer received can handle one more byte for '\0' in case of WS_TEXT data for C string conversion
Not the best optimization code but works.

### CODE REVIEW

File: src/AsyncWebSocket.cpp [hunk 1]
```diff
@@ -725,6 +725,22 @@ void AsyncWebSocket::text(uint32_t id, uint8_t * message, size_t len){
 void AsyncWebSocket::text(uint32_t id, char * message){
   text(id, message, strlen(message));
 }
+void AsyncWebSocket::text(uint32_t id, const __FlashStringHelper *data) {
```
Comment (me-no-dev) [inline src/AsyncWebSocket.cpp]: those method should be added to the AsyncWebSocketClient and pointed to like the other text and binary methods.

File: src/AsyncWebSocket.cpp [hunk 2]
```diff
@@ -725,6 +749,21 @@ void AsyncWebSocket::text(uint32_t id, uint8_t * message, size_t len){
...
+      n += 1;
+  }
+  char * message = (char*) malloc(n+1);
+  if(message){
+    for(size_t b=0; b<n; b++)
+      message[b] = pgm_read_byte(p++);
+    message[n] = 0;
+    text(id, message, n);
+  }
+}
```
Comment (me-no-dev) [inline src/AsyncWebSocket.cpp]: maybe this instead:
```cpp
 void AsyncWebSocket::text(uint32_t id, const __FlashStringHelper *message){
   AsyncWebSocketClient * c = client(id);
   if(c != NULL)
     c->text(message);
 }
 
 void AsyncWebSocket::textAll(const __FlashStringHelper *message){
   AsyncWebSocketClient * c = _clients;
   while(c != NULL){
     if(c->status() == WS_CONNECTED)
       c->text(message);
     c = c->next;
   }
 }
```
same should be done for binary and binaryAll

**Comment by me-no-dev** [comment]
@hallard you around?

**Comment by hallard** [comment]
Yeah in about 1h

**Comment by me-no-dev** [comment]
I would like to merge this when you have time :)

**Comment by hallard** [comment]
no problem, I'm now there and on gitter

**Comment by me-no-dev** [comment]
I would love it if you make the changes I outlined above :) and I will merge it.
The reason I ask for them is because the methods in AsyncWebSocket should only proxy to the client's methods and not actually implement them :) you can see the rest of the methods done that way, with the exception of printf

**Pull request merged**